From 354cd50472093418f8c51e28f9416c35a9c786a8 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Wed, 03 Mar 2021 16:04:31 +0200
Subject: [PATCH] [SW-32407] boot_if: prepare for comms desc

The communication descriptor will be added to the boot_if and will
include types from stdint.h, but it can't include the stdint.h by
itself, so we need to do this for boot_if.
Add stdint.h and a comment about its requirement.

Change-Id: I8fb0b9571212db41da1491646c46a1770afa34f8
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/common/boot_if.c b/common/boot_if.c
index 6badf23..6b4f424 100644
--- a/common/boot_if.c
+++ b/common/boot_if.c
@@ -5,11 +5,12 @@
  */
 
 #include <stdbool.h>
+#include <stdint.h>
 
 #include <psoc_global_conf_regs.h>
 
-#include <hl_boot_if.h>
 #include <preboot.h>
+#include <hl_boot_if.h>			/* requires stdint.h */
 
 /*
  * The cpu_boot_cmd_clr_before_wfe() function should NOT be used in general as
diff --git a/gaudi/common.c b/gaudi/common.c
index ee0b7a6..c8e5007 100644
--- a/gaudi/common.c
+++ b/gaudi/common.c
@@ -5,6 +5,8 @@
  * SPDX-License-Identifier:     GPL-2.0+
  */
 
+#include <stdint.h>
+
 #include <gaudi_blocks.h>
 #include <psoc_global_conf_regs.h>
 #include <psoc_efuse_regs.h>
@@ -19,7 +21,7 @@
 #include <pcie_aux_masks.h>
 #include <pcie_dbi_masks.h>
 
-#include <hl_boot_if.h>
+#include <hl_boot_if.h>			/* requires stdint.h */
 #include <boot_if.h>
 #include <gaudi/gaudi_reg_map.h>
 #include <common/pci_ids.h>
